### PR TITLE
[gen_golden] Support int8 quantization random input

### PR DESCRIPTION
**Pull request by glistening**
gen_golden will generate [-127,127] random input for int8 input type.

ONE-DCO-1.0-Signed-off-by: Sanggyu Lee <sg5.lee@samsung.com>

### CODE REVIEW

**Review by hseok-oh — APPROVED** [review]

File: tools/nnpackage_tool/gen_golden/gen_golden.py [hunk 1]
```diff
@@ -91,6 +91,9 @@ def usage():
             if this_dtype == tf.uint8:
                 input_values.append(
                     np.random.randint(0, 255, this_shape).astype(np.uint8))
+            if this_dtype == tf.int8:
+                input_values.append(
+                    np.random.randint(-127, 127, this_shape).astype(np.int8))
```
Comment (ragmani) [inline tools/nnpackage_tool/gen_golden/gen_golden.py:96]: Maybe -128?

Comment (glistening) [reply tools/nnpackage_tool/gen_golden/gen_golden.py:96]: > Maybe -128?

@ragmani No, we don't need to use the full range of int8 as we did in other type. Also, I considered tensorflow lite uses `[-127, 127]` for weight (symmetric) tensor.

Comment (ragmani) [reply tools/nnpackage_tool/gen_golden/gen_golden.py:96]: > i) we don't need to use the full range of int8 as we did in other type.

`ir::DataType::QUANT_INT8_ASYMM` has range as [-128, 127]. I think we support full range of int to verify this type.

> Too tight range may cause out-of-range outputs as it flows several operators.

Other values like `-127` can also cause out-of-range as it flows several operators. I think out-of-range of outputs is responsible for the kernel of operators. Most of kernels of tensorflow lite handle it.

> ii) Also, I've considered tensorflow lite uses [-127, 127] for weight (symmetric) tensor.

Right for symmetric. But tensorflow lite also uses [-128, 127] for asymmetric tensor. How about supporting both type separately?

Comment (glistening) [reply tools/nnpackage_tool/gen_golden/gen_golden.py:96]: > `ir::DataType::QUANT_INT8_ASYMM` has range as [-128, 127]. I think we support full range of int to verify this type.

First, `gen_golden` does not take input type from nnfw IR, its input is `.tflite`. It just handle `int8` type. Valid input range for `int8` depends on model. I don't want to use tighten the random range which sometimes caused unnecessary errors.

Comment (glistening) [reply tools/nnpackage_tool/gen_golden/gen_golden.py:96]: @ragmani I've just found that I am not the only one who would like to narrow the random input range. Here is the code from tensorflow lite's `benchmark_model`.

```c++
case kTfLiteInt32: {
  int low = has_value_range ? low_range : 0;
  int high = has_value_range ? high_range : 99;
  return CreateInputTensorData<int32_t>(
      num_elements, std::uniform_int_distribution<int32_t>(low, high));
}
...
case kTfLiteInt8: {
  int low = has_value_range ? low_range : -127;
  int high = has_value_range ? high_range : 127;
```

Comment (ragmani) [reply tools/nnpackage_tool/gen_golden/gen_golden.py:96]: > First, gen_golden does not take input type from onert IR, its input is .tflite. It just handle int8 type. Valid input range for int8 depends on model. I don't want to use tight random range which sometimes caused unnecessary errors.

I see. 

> `int8` is not just for `ir::DataType::QUANT_INT8_ASYMM` in gen_golden.py. It should work for ir::DataType::QUANT_INT8_SYMM (not exist at this moment and I am not sure it is necessary).

If `gen_golden.py` generates input of `.tflite`, `int8` input can be `QUANT_INT8_ASYMM` in many cases. But as long as it's not easy to distinguish `int8` in `gen_golden.py`, it seems good to do the range to [-127, 127] what you say.

**Review by ragmani — APPROVED** [review]
LGTM

**Pull request merged**